### PR TITLE
Ft admin get all incidents 162363076

### PR DESCRIPTION
**Pull request by ogiste**
#### What does this PR do?
Adds the endpoint to allow an administrator to view all created incidents

#### Description of Task to be completed?
An administrator should be able to view all incidents created by all users

#### How should this be manually tested?
Clone this repo to your machine

 ``` git clone https://github.com/ogiste/ireporter.git ```

Then change the directory to the project by

``` cd ireporter ```

to make sure all the packages needed to run the project present in your machine,
we'll create a virtual enviroment and install the packages there

* to create a virtual enviroment run

    ``` virtualenv -p python3 venv```
* activating the enviroment

    ``` source venv/bin/activate```

The virtual enviroment is now ready, we can install all packages needed for project
ensure you have pip installed otherwise
then on your terminal run

``` pip install -r requirements.txt ```

# run
To test our project on your terminal run

``` export FLASK_APP=run.py```

then

Export your postgres database url details to the environment:

```
export DB_HOST=localhost
export DB_USER=yourusername
export DB_PASS=yourdbpass
export DB_PORT=5432
export DB_NAME=yourdbname
```

then

``` flask run ```

on your browser open up [http://127.0.0.1:5000/api/v2/incidents/<int:id>](http://127.0.0.1:5000/api/v2/incidents/<int:id>)

# Testing using postman

`GET /api/v2/incidents/all`

#### What are the relevant pivotal tracker stories?
[#162363076](https://www.pivotaltracker.com/story/show/162363076)

### CODE REVIEW

File: app/api/v2/models/incident.py [hunk 1]
```diff
@@ -140,6 +143,48 @@ def get_incidents(self):
...
+
+        dictionary
+            dictionary contaiing an error message and the success status of the update.
+        """
+        incident = self.get_single_incident_by_id(id)
+        if not isinstance(incident, dict):
+            return self.message["NOT_FOUND"] + str(id) \
+             + " Record could not be found or doesnot exist"
+
+        # if incident['status'] != "draft":  # or auth'd user is not admin
```
Comment (kihuha) [inline app/api/v2/models/incident.py]: One of our fellow boot campers was advised by their LFA that comments should be strictly descriptive comments. You should never comment out code but rather, consider removing it all at once if it is not required.

File: app/api/v2/views/incident.py [hunk 2]
```diff
@@ -242,3 +294,107 @@ def delete(self, id, prop=None):
...
+                "msg": incident_data,
+                "status_code": 404
+            }), 404)
+        if isinstance(incident_data, str):
+            return make_response(jsonify({
+                "msg": incident_data,
+                "status_code": 400
+            }), 400)
+        return make_response(
+            jsonify(get_error(error_messages["400"], 400)), 400)
```
[the remark this answers is not in the document]
Comment (ogiste) [reply app/api/v2/views/incident.py]: Sawa, will see what I can do to integrate, thank you

File: app/api/v2/models/incident.py (hunk 1 above)
Comment (ogiste) [reply app/api/v2/models/incident.py]: Definitely! Going to remove this one thanks!

File: app/api/v2/views/incident.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ogiste) [reply app/api/v2/views/incident.py]: It's actually already integrated.

**Pull request merged**